generators/javascript/text.js

```diff
@@ -38,7 +38,7 @@ Blockly.JavaScript['text'] = function(block) {
 Blockly.JavaScript['text_join'] = function(block) {
   // Create a string made up of any number of elements of any type.
   var code;
-  if (block.itemCount_ == 0) {
+  if (block.itemCount_ === 0) {
```

Comment:
I checked the parts of the code where itemCount_ is set and it should never be the case that it is undefined, when this would happen it would be an bug:
...
lists.js (3 matches)
55: this.itemCount_ = 3;  
77: this.itemCount_ = parseInt(xmlElement.getAttribute('items'), 10);  
108: this.itemCount_ = 0; 

text.js (3 matches)
71: this.itemCount_ = 2;  
93: this.itemCount_ = parseInt(xmlElement.getAttribute('items'), 10);  
131: this.itemCount_ = i;  
...

Furthermore I don't found any JSDoc type definition like "@private {!number}" for this variable itemCount_.
So I think the linter message is fine in this case, because the closure compiler would only guess the type without such type definition.

Maybe it would be an good idea to add such type definition for every private definition to make sure that the closure compiler is aware about the type of the different variables to provide smarter warnings.